File: samples/CslaNet/cs/DeepData/DeepData.Library/OrderList.da.cs
using Csla.Data;
using DeepData.DTO;

namespace DeepData.Library
{
	///<summary>
	///</summary>
	public partial class OrderList
	{
		private void FetchDto()
		{
			try
			{
				RaiseListChangedEvents = false;
				IsReadOnly = false;
				DeepData.DAL.DataFactory df = new DeepData.DAL.DataFactory();
				//Dim data() As DeepData.DTO.OrderDto
				DeepData.DTO.OrderDto[] data; //() As Object
				using (DeepData.DAL.OrderData dal = df.GetOrderDataObject())
				{
					//data = CType(dal.GetOrders, DeepData.DTO.OrderDto())
					data = ((DeepData.DTO.OrderDto[])(dal.GetOrders()));
					if (data != null)
					{
						foreach (var order in data)
						{
							Add(OrderInfo.GetOrderInfo(order));
						}
					}
				}
			}
			finally
			{
				IsReadOnly = true;
				RaiseListChangedEvents = true;
			}

		}

	}

}

File: Samples/CslaNet/vb/DeepData/DeepData.Library/OrderList.da.vb
Option Strict Off

Partial Public Class OrderList

  Private Sub FetchDto()

    Try
      Me.RaiseListChangedEvents = False
      IsReadOnly = False
      Dim df As New DeepData.DAL.DataFactory
      'Dim data() As DeepData.DTO.OrderDto
      Dim data '() As Object
      Using dal As DeepData.DAL.OrderData = df.GetOrderDataObject
        'data = CType(dal.GetOrders, DeepData.DTO.OrderDto())
        data = dal.GetOrders
        If data IsNot Nothing Then
          Dim order
          For Each order In data 'As DeepData.DTO.OrderDto In data
            Add(OrderInfo.GetOrderInfo(order))
          Next
        End If
      End Using

    Finally
      IsReadOnly = True
      Me.RaiseListChangedEvents = True
    End Try

  End Sub

End Class
